gitlab/resource_gitlab_topic.go

```diff
@@ -0,0 +1,133 @@
+package gitlab
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strconv"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/xanzy/go-gitlab"
+)
+
+func resourceGitlabTopic() *schema.Resource {
+	return &schema.Resource{
+		Description: "This resource allows you to create and manage topics that are then assignable to projects. Topics are the successors for project tags. Aside from avoiding terminology collisions with Git tags, they are more descriptive and better searchable.\n\nFor assigning topics, use the [project](./project.md) resource.",
```

Comment:
Could you mention that deleting the resource does not delete the topic, using a [markdown callout](https://www.terraform.io/registry/providers/docs#callouts)

Reply:
Should I also just print a warning instead of emptying the fields, according to @wwwjon's [comment](https://github.com/gitlabhq/terraform-provider-gitlab/issues/748#issuecomment-990707136)

Reply:
Yeah I can see that adding a delete feature in the future would be potentially breaking.

How about we add a new required boolean attribute, default false, called `soft_destroy`? Then in the destroy function you run the behavior you already have if the value is true, and return an error if it is false.

Then in the future if topic deletion is supported in the API we can make the attribute optional.